Rectangle/AccessibilityElement.swift

```diff
@@ -70,6 +81,43 @@ class AccessibilityElement {
     }
     
     static func getWindowInfo(at location: CGPoint) -> CGWindowInfo? {
+        let options = CGWindowListOption(arrayLiteral: .excludeDesktopElements, .optionOnScreenOnly)
+        if let windowInfo = CGWindowListCopyWindowInfo(options, 0) as? Array<Dictionary<String,Any>> {
+            var windowInfos = [CGWindowInfo]()
+            for infoDict in windowInfo {
+                if let bounds = infoDict[kCGWindowBounds as String] as? [String: CGFloat] {
+                    guard let pid = infoDict[kCGWindowOwnerPID as String] as? pid_t,
+                          let id = infoDict[kCGWindowNumber as String] as? Int,
+                          let x = bounds["X"],
+                          let y = bounds["Y"],
+                          let w = bounds["Width"],
+                          let h = bounds["Height"]
+                    else { continue }
+                    let name = infoDict[kCGWindowOwnerName as String] as? String
+                    if name != "Dock" && name != "WindowManager" {
+                        let boundsRect = NSMakeRect(x, y, w, h)
+                        windowInfos.append(CGWindowInfo(id: id, pid: pid, rect: boundsRect))
+                    }
+                }
+            }
+            if var resultWindowInfo = windowInfos.first(where: { $0.rect.contains(location) }) {
+                if StageUtil.stageCapable() && StageUtil.stageEnabled() && StageUtil.stagePresent(windowInfo) {
+                    // In case the window is in Stage Manager recent apps
+                    var prevWindowInfo: CGWindowInfo?
+                    while prevWindowInfo?.id != resultWindowInfo.id {
```

Comment:
I feel like I'm missing something here with this while loop. It appears to me that it is unnecessary, but perhaps you can elaborate on what it's doing?

Reply:
You can only drag the foreground window of a scene from the recent apps. This loop is used to find this window.

Reply:
Hmm. Sorry to keep pushing on this, but when I test this out, I never get a different value for the resultWindowInfo (there's no iteration in the while loop because it gets the same value twice and exits, and it's not straightforward to me how it would obtain different values in any different scenario). Even if I remove this block entirely, I can still drag & snap the windows from recent apps because it appears to get the correct windowId. Perhaps there is an edge case that I'm not hitting that would justify having this?

The reason why this caught my attention in the first place is because 1) while loops that don't have an intuitive exit are too easy to end up in an infinite loop (if not now, maybe with other additional refactoring later), and 2) I pretty much never force unwrap optionals. Both of these scenarios are potential for application crashes, and even if your code here was straightforward to me, I would have still suggested removing both of those. In other apps I've written, places where I was certain there was no chance of a force unwrap causing a crash, still caused a crash due to some very tricky memory management from macOS that had released memory I thought wouldn't get released. One could argue that a crash is desirable in certain scenarios, but I rarely have those scenarios - I'd almost always want the app to still chug along even if there's something like this that isn't quite right.

Reply:
Thanks for the clarification, I understand now. As for the while loop, I'm good with it as-is. I appreciate that you've continued the work on Stage Manager!